tests/test_tokenization_common.py

```diff
@@ -1153,6 +1153,120 @@ def test_chat_template_batched(self):
+                        )
+                        # assert labels in assistant indices
+                        self.assertEqual(
+                            output["assistant_masks"][i][assistant_start : assistant_end + 1],
+                            [1] * (assistant_end - assistant_start + 1),
+                        )
+                        self.assertEqual(
+                            output["assistant_masks"][i][assistant_start2 : assistant_end2 + 1],
+                            [1] * (assistant_end2 - assistant_start2 + 1),
+                        )
+                        # assert 0 in user/system indices
+                        self.assertEqual(all(l == 0 for l in output["assistant_masks"][i][:assistant_start]), True)
+                        self.assertEqual(
+                            all(l == 0 for l in output["assistant_masks"][i][assistant_end + 1 : assistant_start2]),
+                            True,
+                        )
+
+                    # check not batched
+                    output = tokenizer_r.apply_chat_template(
+                        conversations[0],
+                        chat_template=dummy_template,
+                        tokenize=True,
+                        return_assistant_tokens_mask=True,
+                        return_dict=True,
+                    )
+
+                    chat_string = tokenizer_r.apply_chat_template(
+                        conversations[0], tokenize=False, chat_template=dummy_template
+                    )
+                    assistant_start = output.char_to_token(0, chat_string.index(assistant_wrappers[0][0][0]))
+                    assistant_end = output.char_to_token(
+                        0, chat_string.index(assistant_wrappers[0][0][1]) + len(assistant_wrappers[0][0][1]) - 1
+                    )
+                    assistant_start2 = output.char_to_token(0, chat_string.index(assistant_wrappers[0][1][0]))
+                    assistant_end2 = output.char_to_token(
+                        0, chat_string.index(assistant_wrappers[0][1][1]) + len(assistant_wrappers[0][1][1]) - 1
+                    )
+
+                    # assert labels in assistant indices
+                    self.assertEqual(
+                        output["assistant_masks"][assistant_start : assistant_end + 1],
+                        [1] * (assistant_end - assistant_start + 1),
+                    )
+                    self.assertEqual(
+                        output["assistant_masks"][assistant_start2 : assistant_end2 + 1],
+                        [1] * (assistant_end2 - assistant_start2 + 1),
+                    )
+
+                    # assert -100 in user/system indices
+                    self.assertEqual(all(l == 0 for l in output["assistant_masks"][:assistant_start]), True)
+                    self.assertEqual(
+                        all(l == 0 for l in output["assistant_masks"][assistant_end + 1 : assistant_start2]), True
+                    )
```

Comment:
* It's better to return early and have the bulk of logic indented as little as possible
* If skipping, this should be done explicitly, otherwise we think things are passing when they're not 


```suggestion
                if not self.test_rust_tokenizer:
                    self.skipTest(reason="No fast tokenizer defined")
                
                tokenizer_r = self.rust_tokenizer_class.from_pretrained(pretrained_name)

                # check batched
                output = tokenizer_r.apply_chat_template(
                    conversations,
                    chat_template=dummy_template,
                    tokenize=True,
                    return_assistant_tokens_mask=True,
                    return_dict=True,
                )
                for i, conv in enumerate(conversations):
                    chat_string = tokenizer_r.apply_chat_template(
                        conversations[i], tokenize=False, chat_template=dummy_template
                    )
                    assistant_start = output.char_to_token(i, chat_string.index(assistant_wrappers[i][0][0]))
                    assistant_end = output.char_to_token(
                        i, chat_string.index(assistant_wrappers[i][0][1]) + len(assistant_wrappers[i][0][1]) - 1
                    )

                    assistant_start2 = output.char_to_token(i, chat_string.index(assistant_wrappers[i][1][0]))
                    assistant_end2 = output.char_to_token(
                        i, chat_string.index(assistant_wrappers[i][1][1]) + len(assistant_wrappers[i][1][1]) - 1
                    )
                    # assert labels in assistant indices
                    self.assertEqual(
                        output["assistant_masks"][i][assistant_start : assistant_end + 1],
                        [1] * (assistant_end - assistant_start + 1),
                    )
                    self.assertEqual(
                        output["assistant_masks"][i][assistant_start2 : assistant_end2 + 1],
                        [1] * (assistant_end2 - assistant_start2 + 1),
                    )
                    # assert 0 in user/system indices
                    self.assertEqual(all(l == 0 for l in output["assistant_masks"][i][:assistant_start]), True)
                    self.assertEqual(
                        all(l == 0 for l in output["assistant_masks"][i][assistant_end + 1 : assistant_start2]),
                        True,
                    )

                # check not batched
                output = tokenizer_r.apply_chat_template(
                    conversations[0],
                    chat_template=dummy_template,
                    tokenize=True,
                    return_assistant_tokens_mask=True,
                    return_dict=True,
                )

                chat_string = tokenizer_r.apply_chat_template(
                    conversations[0], tokenize=False, chat_template=dummy_template
                )
                assistant_start = output.char_to_token(0, chat_string.index(assistant_wrappers[0][0][0]))
                assistant_end = output.char_to_token(
                    0, chat_string.index(assistant_wrappers[0][0][1]) + len(assistant_wrappers[0][0][1]) - 1
                )
                assistant_start2 = output.char_to_token(0, chat_string.index(assistant_wrappers[0][1][0]))
                assistant_end2 = output.char_to_token(
                    0, chat_string.index(assistant_wrappers[0][1][1]) + len(assistant_wrappers[0][1][1]) - 1
                )

                # assert labels in assistant indices
                self.assertEqual(
                    output["assistant_masks"][assistant_start : assistant_end + 1],
                    [1] * (assistant_end - assistant_start + 1),
                )
                self.assertEqual(
                    output["assistant_masks"][assistant_start2 : assistant_end2 + 1],
                    [1] * (assistant_end2 - assistant_start2 + 1),
                )

                # assert -100 in user/system indices
                self.assertEqual(all(l == 0 for l in output["assistant_masks"][:assistant_start]), True)
                self.assertEqual(
                    all(l == 0 for l in output["assistant_masks"][assistant_end + 1 : assistant_start2]), True
                )
```